src/sparse_matrix.cpp

```diff
@@ -86,10 +94,19 @@ void CSRMatrix::set(unsigned i, unsigned j, const RCP<const Basic> &e)
 
     unsigned k = p_[i];
     unsigned row_end = p_[i + 1];
+    unsigned mid = k;
 
-    // TODO: Use binary search
-    while (k < row_end && j_[k] < j)
-        k++;
+    while (k < row_end) {
```

Comment:
Wrote the search keeping in mind the post conditions.
The tests fail for this but pass for the other two commits.